osquery/dispatcher/scheduler.cpp

```diff
@@ -109,24 +110,24 @@ void launchQuery(const std::string& name, const ScheduledQuery& query) {
 }
 
 void SchedulerRunner::start() {
+  VLOG(1) << "Starting the schedule runner";
   time_t t = std::time(nullptr);
   struct tm* local = std::localtime(&t);
   unsigned long int i = local->tm_sec;
   for (; (timeout_ == 0) || (i <= timeout_); ++i) {
-    {
-      ConfigDataInstance config;
-      for (const auto& query : config.schedule()) {
-        if (i % query.second.splayed_interval == 0) {
-          launchQuery(query.first, query.second);
-        }
-      }
-    }
+    Config::getInstance().scheduledQueries(
+        ([&i](const std::string& name, const ScheduledQuery& query) {
+          if (i % query.splayed_interval == 0) {
```

Comment:
It is worth adding a `1` to the interval like before? In the rare and unfortunate case that a splay or interval ever becomes `0`. Otherwise an inline check here for a `0` interval?